pac4j-saml/src/main/java/org/pac4j/saml/metadata/SAML2ServiceProviderMetadataResolver.java

```diff
@@ -45,28 +44,27 @@
     private final boolean forceSpMetadataGeneration;
     private boolean authnRequestSigned;
     private boolean wantsAssertionsSigned;
+    private String binding;
 
-    public SAML2ServiceProviderMetadataResolver(final SAML2ClientConfiguration configuration,
-                                                final String callbackUrl,
+    public SAML2ServiceProviderMetadataResolver(final SAML2ClientConfiguration configuration, final String callbackUrl,
                                                 final CredentialProvider credentialProvider) {
         this(configuration.getServiceProviderMetadataResource(), callbackUrl,
                 configuration.getServiceProviderEntityId(), configuration.isForceServiceProviderMetadataGeneration(), credentialProvider,
-                configuration.isAuthnRequestSigned(), configuration.getWantsAssertionsSigned());
+                configuration.isAuthnRequestSigned(), configuration.getWantsAssertionsSigned(), configuration.getDestinationBindingType());
     }
 
-    private SAML2ServiceProviderMetadataResolver(final WritableResource spMetadataResource,
-                                                 final String callbackUrl,
-                                                 @Nullable final String spEntityId,
-                                                 final boolean forceSpMetadataGeneration,
-                                                 final CredentialProvider credentialProvider,
-                                                 boolean authnRequestSigned, boolean wantsAssertionsSigned) {
+    private SAML2ServiceProviderMetadataResolver(final WritableResource spMetadataResource, final String callbackUrl,
```

Comment:
This all looks good. 

Question: Do we really need a constructor to pass bits and pieces from `configuration` individually?

Reply:
Yes, indeed, the private constructor is useless. Will remove it.